src/features/unified-share-modal/__tests__/EmailForm.test.js

```diff
@@ -311,14 +311,42 @@ describe('features/unified-share-modal/EmailForm', () => {
                 expectedValue: false,
             },
         ].forEach(({ email, expectedValue }) => {
-            test('should validate email addresses properly', () => {
+            test('should properly validate pill text input as email address', () => {
                 const wrapper = getWrapper();
 
-                const isValidEmail = wrapper.instance().isValidEmail(email);
+                const isValidContactPill = wrapper.instance().isValidContactPill(email);
 
-                expect(isValidEmail).toBe(expectedValue);
+                expect(isValidContactPill).toBe(expectedValue);
             });
         });
+
+        test('should consider parsed contact pills as valid by default', () => {
+            const wrapper = getWrapper();
+
+            expectedContacts.forEach(contact => {
+                const isValidContactPill = wrapper.instance().isValidContactPill(contact);
+
+                expect(isValidContactPill).toBe(true);
+            });
+        });
+
+        test('should invalidate parsed contact pills when shouldInvalidateExternalCollabs is true and contact is an external user', () => {
```

Comment:
The description only refers to the case in which `shouldInvalidateExternalCollabs` is `true`, but this test also covers the case in which `shouldInvalidateExternalCollabs` is `false`. Should it be split into two tests in a `test.each` loop?